src/ParallelEnv.f90

```diff
@@ -24,25 +24,38 @@ MODULE ParallelEnv
 USE Allocs
 !$ USE OMP_LIB
 
+#ifdef FUTILITY_HAVE_PETSC
+#include <petscversion.h>
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
+USE PETSCSYS, ONLY: PETSC_NULL_CHARACTER
+#endif
+#endif
+
 IMPLICIT NONE
 PRIVATE
 
 #ifdef HAVE_MPI
-
 #ifdef FUTILITY_HAVE_PETSC
-#include <petscversion.h>
 #if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>=6))
 #include <petsc/finclude/petsc.h>
 #else
 #include <finclude/petsc.h>
 #endif
 #undef IS
-PetscErrorCode  :: ierr
-PetscBool :: petsc_isinit
+#if ((PETSC_VERSION_MAJOR>=3) && (PETSC_VERSION_MINOR>6))
+INCLUDE 'mpif.h'
+#endif
 #else
 INCLUDE 'mpif.h'
 #endif
+#endif
 
+#ifdef FUTILITY_HAVE_PETSC
+PetscErrorCode  :: ierr
+PetscBool :: petsc_isinit
+#endif
```

Comment:
Any reason this had to move down to its own section?

Reply:
preference